enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+
+3. What happens when someone updates the eipAllocations for the load balancer service of an Ingress Controller ?
+
+It is not supported to manually edit the eipAllocations for the load balancer service of an Ingress Controller.
+However, if someone does that then they will find the error in status as follows in the IngressController:
+```yaml
+  - lastTransitionTime: "2024-04-17T03:09:32Z"
+    message: |-
+      The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: error creating load balancer: "ResourceInUse: The allocation IDs are not available for use\n\tstatus code: 400, request id: 58136c42-e1bd-4c1a-ac38-7078da2b1d95"
+      The kube-controller-manager logs may contain more details.
+    reason: SyncLoadBalancerFailed
+    status: "False"
+    type: LoadBalancerReady
+  - lastTransitionTime: "2024-04-17T02:54:25Z"
+    message: LoadBalancer is not progressing
+    reason: LoadBalancerNotProgressing
+    status: "False"
+    type: LoadBalancerProgressing
+...
+...
+  - lastTransitionTime: "2024-04-17T03:09:32Z"
+    message: |-
+      One or more status conditions indicate unavailable: LoadBalancerReady=False (SyncLoadBalancerFailed: The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: error creating load balancer: "ResourceInUse: The allocation IDs are not available for use\n\tstatus code: 400, request id: 58136c42-e1bd-4c1a-ac38-7078da2b1d95"
+      The kube-controller-manager logs may contain more details.)
+    reason: IngressControllerUnavailable
+    status: "False"
+    type: Available
+
+```
+
+- Are there any pre-requisites for this EP ?
+  Yes, one must [allocate Elastic IP address](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating)
+  in their AWS environment which could be then used in the `eipAllocations` field of the IngressController CR.
+
+- What happens when the cluster-admin specified invalid EIPs in install-config.yaml at installation time (day 0) ?
+  We would try to get the installer feasibly validate the install config for day 0. However, we still need verify this
+  with the installer team. Mostly, we will need to add validations in [pkg/types/aws/validation/platform.go](https://github.com/openshift/installer/blob/master/pkg/types/aws/validation/platform.go)
+
+Note: The following questions are applicable when `auto-delete-load-balancer` annotation is set in the IngressController as it tries to recreate the update Load Balancer.
+If not set then the LoadBalancerProgressing Status will be set to "True" with an message as described in [subsection of Test Plan](#update-an-ingress-controller-eipallocations-by-adding-auto-delete-load-balancer-annotation-in-the-ingress-controller-cr-which-already-has-eipallocations-set)
```

Comment:
Which following questions is this talking about? You mean `What happens when a cluster-admin adds invalid EIPs ?`,  `What happens when a cluster-admin adds number of EIPs...`, and `- What happens when a cluster-admin adds nothing under eipAllocations ?`?

I don't understand. If you don't have `auto-delete-load-balancer` set, then you will get a `LoadBalancerProgressing` Status, yes I agree, but that status won't tell them the EIPs are invalid.

They are still going to proceed with deleting the service, and then the service will fail to provision. So what's the point of adding this predicate note of `The following questions...`? I don't think it changes anything right?

Reply:
Yes.